app/helpers/argo_helper.rb

```diff
@@ -179,7 +179,7 @@ def render_foxml_link(document, link_text = 'FoXML', opts = {:target => '_blank'
   end
 
   def render_solr_link(document)
-    return '' unless current_user.is_admin
+    return '' unless can? :read, document
```

Comment:
Not clear whether this will behave the same way and is less clear about it's intent.  It could be that this should not change.